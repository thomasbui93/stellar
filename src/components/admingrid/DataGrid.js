import React from 'react'
import {DataRow} from './DataRow'

export const DataGrid = props => {
  const {items, rowClassName, baseUrl, className, removeAction} = props;
  return <div className={`admin-data-grid ${className}`}>
    {items && items.length > 0 ? renderList(items, rowClassName, baseUrl, removeAction) : <p className="empty-grid">Your grid is currently empty</p>}
  </div>
};

const renderList = (items, rowClassName, baseUrl, removeAction) => {
  return items.map(item => {
    const populatedItem = {...item, baseUrl, removeAction};
    return <DataRow item={populatedItem} className={rowClassName} key={item.key} />
  })
};
