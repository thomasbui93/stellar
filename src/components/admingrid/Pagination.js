import React from 'react'
import PropTypes from 'prop-types'

const Pagination = ({next, prev, requestApi}) => {
  return (
    <div className='pagination'>
      { next > 0 ? <div className='pagination-next' onClick={() => { requestApi({page: next}) }}>Next</div> : ''}
      { prev > 0 ? <div className='pagination-previous' onClick={() => { requestApi({page: prev}) }}>Prev</div> : ''}
    </div>
  )
}

Pagination.propTypes = {
  next: PropTypes.number,
  prev: PropTypes.number,
  requestApi: PropTypes.func
}

export { Pagination }
