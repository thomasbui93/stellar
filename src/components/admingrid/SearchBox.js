import React, {Component} from 'react'
import PropTypes from 'prop-types'

export class SearchBox extends Component {
  static propTypes = {
    requestApi: PropTypes.func
  };

  constructor(props) {
    super(props);
    this.submitAction = this.submitAction.bind(this);
    this.clearSearch = this.clearSearch.bind(this);
  }

  submitAction(event) {
    event.preventDefault();
    const query = this.refs.query.value;
    if (query.length >= 3) {
      this.props.requestApi({query: query})
    }
  }

  clearSearch(){
    this.refs.query.value = '';
    this.props.requestApi();
  }

  render() {
    return <form className={`search-box ${this.props.className ? this.props.className : ''}`}
                 onSubmit={this.submitAction}>
      <input placeholder="Enter your search here" type="text" ref='query'/>
      <button type="submit">Search</button>
      <button type="button" onClick={this.clearSearch}>
        Clear
      </button>
    </form>
  }
}