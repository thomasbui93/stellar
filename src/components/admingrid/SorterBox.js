import React, {Component} from 'react'
import PropTypes from 'prop-types'

export class SorterBox extends Component {
  constructor (props) {
    super(props)
    this.applySortAction = this.applySortAction.bind(this)
  }

  static propTypes = {
    requestApi: PropTypes.func,
    fields: PropTypes.arrayOf(PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.label
    }))
  };

  applySortAction () {
    this.props.requestApi({sort: this.refs.field.value, desc: this.refs.dir.value})
  }

  render () {
    const {fields} = this.props
    return <div className='sorter-box'>
      {fields.length > 0
        ? <div>
          <select className='sorter-box__field' ref='field' onChange={this.applySortAction}>
            { fields.map(({value, label}) => {
              return <option value={value} key={value}> {label}</option>
            })}
          </select>
          <select className='sorter-box__direction' ref='dir' onChange={this.applySortAction}>
            <option value={'true'} >DESC</option>
            <option value={'false'}>ASC</option>
          </select>
        </div>
        : ''
      }
    </div>
  }
}
