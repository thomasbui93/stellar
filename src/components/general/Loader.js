import React from 'react'

export const Loader = () => {
  return <div className='app-loader'>
    <div className='loader' />
    <div className='loader-text'>Loading...</div>
  </div>
}
