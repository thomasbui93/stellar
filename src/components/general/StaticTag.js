import React from 'react';

export const StaticTag = (props) => {
  return <div className={`${props.className} static-tag`}>{props.staticText}</div>
};