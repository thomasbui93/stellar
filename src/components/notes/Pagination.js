import React from 'react'
import PropTypes from 'prop-types'

export default class NoteThumbItem extends React.Component {
  static propTypes = {
    numberOfPages: PropTypes.number,
    navigate: PropTypes.func,
    currentPage: PropTypes.number
  }

  render () {
    const paginationSequences = Array.from({length: this.props.numberOfPages}, (v, k) => k + 1)
    return (
      <div className='pagination'>
        {
          paginationSequences.length > 1
          ? paginationSequences.map(sequence => {
            return <div
              className={`${this.props.currentPage === sequence ? 'current-page' : ''} pagination-page`}
              key={sequence}
              onClick={() => this.props.navigate(sequence)}> {sequence} </div>
          }) : ''
        }
      </div>
    )
  }
}
