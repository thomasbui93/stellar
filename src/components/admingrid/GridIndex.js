import React, {Component} from 'react'
import PropTypes from 'prop-types'
import {DataGrid} from './DataGrid'
import {StaticTag} from '../general/StaticTag'
import {Loader} from '../../components/general/Loader'
import {Pagination} from './Pagination'
import {SearchBox} from './SearchBox'
import {SorterBox} from './SorterBox'

export class GridIndex extends Component {
  static propTypes = {
    items: PropTypes.array,
    isLoading: PropTypes.bool,
    removeAction: PropTypes.func,
    requestApi: PropTypes.func,
    sortingFields: PropTypes.arrayOf(PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string
    })),
    pagination: PropTypes.shape({
      next: PropTypes.number,
      prev: PropTypes.number
    }),
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
  };

  componentDidMount () {
    this.props.requestApi()
  }

  render () {
    const {
      error,
      isLoading,
      items,
      removeAction,
      match,
      pagination,
      requestApi,
      sortingFields
    } = this.props

    return <div className={this.props.className}>
      <div className='messages'>
        {error ? <StaticTag className='request-error' staticText={error} /> : ''}
      </div>
      <div className='filter-box'>
        <SearchBox requestApi={requestApi} />
        <SorterBox requestApi={requestApi} fields={sortingFields} />
      </div>
      <Pagination requestApi={requestApi} next={pagination.next} prev={pagination.prev} />
      <div>
        {isLoading ? <Loader /> : ''}
        <DataGrid
          baseUrl={match.url}
          items={items}
          removeAction={removeAction} />
      </div>
      <Pagination requestApi={requestApi} next={pagination.next} prev={pagination.prev} />
    </div>
  }
}
