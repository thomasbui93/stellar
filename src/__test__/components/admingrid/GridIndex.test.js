import React from 'react'
import renderer from 'react-test-renderer'
import {GridIndex} from '../../../components/admingrid/GridIndex'
import {MemoryRouter, Route} from 'react-router';
import {mount} from 'enzyme'
import {expect as expectChai} from 'chai'
import {SearchBox} from "../../../components/admingrid/SearchBox"
import {SorterBox} from "../../../components/admingrid/SorterBox"
import {DataGrid} from "../../../components/admingrid/DataGrid"
import {Pagination} from "../../../components/admingrid/Pagination"
import {Loader} from "../../../components/general/Loader";
import {StaticTag} from "../../../components/general/StaticTag";

describe('GridIndex', () => {
  const removeAction = () => {
    console.log('remove action');
  };

  const requestApi = () => {
    console.log('request api');
  };

  const sortingFields = [{
    value: 'name',
    label: 'Name'
  }, {
    value: 'description',
    label: 'Description'
  }];

  const items = [{
    createdAt: "2017-10-12T15:58:45.164Z",
    description: "Test Tag Description 3",
    isActive: true,
    key: "snake-S16NWzphW",
    name: "Snake",
    updatedAt: "2017-10-12T15:58:45.164Z"
  }, {
    createdAt: "2017-11-12T15:58:45.164Z",
    description: "Test Tag Description 4",
    isActive: true,
    key: "snake-S26NWzphW",
    name: "Snake 1",
    updatedAt: "2017-08-12T15:58:45.164Z"
  }];
  const pagination = {
    next: 3,
    prev: 1
  };

  it('render correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Route exact path="/" render={ props =>{
          return <GridIndex
            error={false}
            isLoading={false}
            requestApi={requestApi}
            removeAction={removeAction}
            sortingFields={sortingFields}
            items={items}
            pagination={pagination}
            location={props.location}
            match={props.match}
          />
        }}>
        </Route>
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render correct sub components', () => {
    const componentNormal = mount(
      <MemoryRouter>
        <Route exact path="/" render={ props =>{
          return <GridIndex
            error={false}
            isLoading={false}
            requestApi={requestApi}
            removeAction={removeAction}
            sortingFields={sortingFields}
            items={items}
            pagination={pagination}
            location={props.location}
            match={props.match}
          />
        }}>
        </Route>
      </MemoryRouter>
    );

    expectChai(componentNormal.find(SearchBox)).to.have.length(1);
    expectChai(componentNormal.find(SorterBox)).to.have.length(1);
    expectChai(componentNormal.find(Pagination)).to.have.length(2);
    expectChai(componentNormal.find(DataGrid)).to.have.length(1);

    const componentLoading = mount(
      <MemoryRouter>
        <Route exact path="/" render={ props =>{
          return <GridIndex
            error={false}
            isLoading={true}
            requestApi={requestApi}
            removeAction={removeAction}
            sortingFields={sortingFields}
            items={items}
            pagination={pagination}
            location={props.location}
            match={props.match}
          />
        }}>
        </Route>
      </MemoryRouter>
    );
    expectChai(componentLoading.find(Loader)).to.have.length(1);

    const componentError = mount(
      <MemoryRouter>
        <Route exact path="/" render={ props =>{
          return <GridIndex
            error={'Sample error'}
            isLoading={false}
            requestApi={requestApi}
            removeAction={removeAction}
            sortingFields={sortingFields}
            items={items}
            pagination={pagination}
            location={props.location}
            match={props.match}
          />
        }}>
        </Route>
      </MemoryRouter>
    );
    expectChai(componentError.find(StaticTag)).to.have.length(1);
  })
});