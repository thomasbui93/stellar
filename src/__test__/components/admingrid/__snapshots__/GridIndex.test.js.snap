// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridIndex render correctly 1`] = `
<div
  className={undefined}
>
  <div
    className="messages"
  >
    
  </div>
  <div
    className="filter-box"
  >
    <form
      className="search-box "
      onSubmit={[Function]}
    >
      <input
        placeholder="Enter your search here"
        type="text"
      />
      <button
        type="submit"
      >
        Search
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        Clear
      </button>
    </form>
    <div
      className="sorter-box"
    >
      <div>
        <select
          className="sorter-box__field"
          onChange={[Function]}
        >
          <option
            value="name"
          >
             
            Name
          </option>
          <option
            value="description"
          >
             
            Description
          </option>
        </select>
        <select
          className="sorter-box__direction"
          onChange={[Function]}
        >
          <option
            value="true"
          >
            DESC
          </option>
          <option
            value="false"
          >
            ASC
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <div
      className="pagination-next"
      onClick={[Function]}
    >
      Next
    </div>
    <div
      className="pagination-previous"
      onClick={[Function]}
    >
      Prev
    </div>
  </div>
  <div>
    
    <div
      className="admin-data-grid undefined"
    >
      <div
        className="admin-grid-row"
      >
        <div
          className="admin-grid-row__name"
        >
          Snake
        </div>
        <div
          className="admin-grid-row__key"
        >
          snake-S16NWzphW
        </div>
        <div
          className="admin-grid-row__status"
        >
          Active
        </div>
        <div
          className="admin-grid-row__action"
        >
          <div
            className="admin-grid-row__remove"
            onClick={[Function]}
          >
            Delete
          </div>
          <a
            className="admin-grid-row__edit"
            href="//snake-S16NWzphW"
            onClick={[Function]}
          >
            Edit
          </a>
        </div>
      </div>
      <div
        className="admin-grid-row"
      >
        <div
          className="admin-grid-row__name"
        >
          Snake 1
        </div>
        <div
          className="admin-grid-row__key"
        >
          snake-S26NWzphW
        </div>
        <div
          className="admin-grid-row__status"
        >
          Active
        </div>
        <div
          className="admin-grid-row__action"
        >
          <div
            className="admin-grid-row__remove"
            onClick={[Function]}
          >
            Delete
          </div>
          <a
            className="admin-grid-row__edit"
            href="//snake-S26NWzphW"
            onClick={[Function]}
          >
            Edit
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <div
      className="pagination-next"
      onClick={[Function]}
    >
      Next
    </div>
    <div
      className="pagination-previous"
      onClick={[Function]}
    >
      Prev
    </div>
  </div>
</div>
`;
